templates/admin.html

```
@@ -736,6 +736,147 @@ <h2>SMTP settings</h2>
+       "API token"), so adding a gateway in src/sms/factory.rs is enough. #}
+    {% for p in sms_providers %}
+    <div class="sms-provider-fields" data-provider="{{ p.kind }}" style="display: {% if sms_provider == p.kind %}block{% else %}none{% endif %};">
+      {% if p.api_key_label %}
+      <div class="form-group">
+        <label>{{ p.api_key_label }}</label>
+        <input type="text" name="api_key" value="{% if sms_provider == p.kind %}{{ sms_api_key }}{% endif %}"{% if sms_provider != p.kind %} disabled{% endif %}>
+      </div>
+      {% endif %}
+      <div class="form-group">
+        <label>{{ p.api_secret_label }} <span class="hint">(leave empty to keep current)</span></label>
+        <input type="password" name="api_secret" value="" placeholder="{% if sms_provider == p.kind %}Leave empty to keep unchanged{% else %}Required when switching gateway{% endif %}"{% if sms_provider != p.kind %} disabled{% endif %}>
```

HTMLHint (1.9.2)

[error] Special characters must be escaped : [ < ].

(spec-char-escape)